api/src/exam-environment/routes/exam-environment.test.ts

```diff
@@ -330,10 +335,14 @@ describe('/exam-environment/', () => {
       });
 
       it('should return an error if the exam has been attempted too recently to retake', async () => {
+        const examTotalTimeInMS = mock.exam.config.totalTimeInS
+          ? mock.exam.config.totalTimeInS * 1000
+          : mock.exam.config.totalTimeInMS;
```

Comment:
```suggestion
        const examTotalTimeInMS = mock.exam.config.totalTimeInS;
```
Tests should not be conditional. `mock.exam.config.totalTimeInS` is an input you control, so you don't need to check it in the test.

We can use `satisfies` to make TS happy. e.g. `const config = {...} satisfies ExamEnvironmentConfig;` and so on.

There's a lot though, so I'll push a commit.

Reply:
That is nice. Thank you.